table/table.go

```diff
@@ -78,7 +78,7 @@ type Table interface {
 	Truncate(ctx context.Context) (err error)
 
 	// AddRecord inserts a row into the table.
-	AddRecord(ctx context.Context, r []interface{}) (recordID int64, err error)
+	AddRecord(ctx context.Context, r []interface{}, h int64) (recordID int64, err error)
```

Comment:
And how about move generating auto id `recordID, err = t.alloc.Alloc(t.ID)` out of AddRecord function?